src/mlpack/tests/ada_grad_test.cpp

```diff
@@ -0,0 +1,107 @@
+/**
+ * @file ada_grad_test.cpp
+ * @author Abhinav Moudgil
+ *
+ * Test file for AdaGrad (stochastic gradient descent with AdaGrad updates).
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#include <mlpack/core.hpp>
+#include <mlpack/core/optimizers/sgd/sgd.hpp>
```

Comment:
I think we can remove the header file here. We include ``SGDTestFunction`` with ``#include <mlpack/core/optimizers/sgd/test_function.hpp>``.